utils/kubernetes/volumes.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2023 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubernetes
+
+import (
+	v1 "k8s.io/api/core/v1"
+)
+
+func ConfigMapsToVolumesAndMount(configMaps map[string]v1.ConfigMap, prefixMountPathMap map[string]string) ([]v1.Volume, []v1.VolumeMount) {
+	volumes := make([]v1.Volume, 0)
+	volumeMounts := make([]v1.VolumeMount, 0)
+	for keycm, cm := range configMaps {
+		volumes = append(volumes, Volume(cm.Name, cm.Name))
+		volumeMounts = append(volumeMounts, VolumeMount(cm.Name, true, prefixMountPathMap[keycm]))
```

Comment:
what assures that prefixMountPathMap[keycm] is always populated with the with the configmap key?

Reply:
Is a configuration map  created by us here https://github.com/kiegroup/kogito-serverless-operator/pull/81/files#diff-cc635941ea89d2ed576d6a3dc9e4c926858209209ae85f93d3502120369c432aR52-R57 if is not found a VolumeMount will not be appended

Reply:
If the confimpaps aren't present the related volume and volumemount will not be mounted

Reply:
will the user be notified somehow?
at least a log message?

Reply:
@spolti  Yes here https://github.com/kiegroup/kogito-serverless-operator/pull/81/files#diff-cc635941ea89d2ed576d6a3dc9e4c926858209209ae85f93d3502120369c432aR157